android/app/src/main/res/layout/max_widget_layout.xml

```diff
@@ -0,0 +1,427 @@
+
+            <TextView
+                android:id="@+id/locationRegionTextView"
+                style="@style/locationRegion"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="center_horizontal"
+                android:gravity="center_horizontal"
+                android:textSize="15sp"
+                tools:text="Suomi" />
+
+        </LinearLayout>
+
+        <TextView
+            android:id="@+id/timeTextView"
+            style="@style/currentWeatherTime"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:gravity="center_horizontal"
+            android:textSize="15sp"
+            tools:text="klo 11:00" />
+
+        <LinearLayout
+            android:id="@+id/weatherLayout"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:layout_marginTop="3dp"
+            android:layout_marginBottom="7dp"
+            android:orientation="horizontal">
+
+            <ImageView
+                android:id="@+id/weatherIconImageView"
+                android:layout_width="46dp"
+                android:layout_height="46dp"
+                android:layout_marginEnd="18dp"
+                android:contentDescription="TODO"
+                tools:src="@drawable/s32_light" />
+
+            <TextView
+                android:id="@+id/temperatureTextView"
+                style="@style/temperature"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="center_vertical"
+                tools:text="+12" />
+
+            <TextView
+                android:id="@+id/temperatureUnitTextView"
+                style="@style/temperatureUnit"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="6dp"
+                tools:text="°C" />
+
+        </LinearLayout>
+
+        <RelativeLayout
+            android:id="@+id/hourForecastRowLayout"
+            android:layout_width="wrap_content"
+            android:layout_height="80dp"
+            android:orientation="horizontal">
+
+            <ImageView
+                android:layout_width="300dp"
+                android:layout_height="20dp"
+                android:background="#F6F6F7"
+                android:importantForAccessibility="no" />
+
+            <ImageView
+                android:layout_width="300dp"
+                android:layout_height="20dp"
+                android:layout_alignParentBottom="true"
+                android:background="#F6F6F7"
+                android:importantForAccessibility="no" />
+
+            <RelativeLayout
```

Comment:
It is not good way to hard code multiple RelativeLayouts. You should add these columns programmatically.

Reply:
To be more clear (because I put my comment on wrong line), I think that it is OK to have the first column with constant icons here (@+id/weatherIconsLayout). But repeated RelativeLayouts with indexes like "@+id/weatherLayout0" should be handled programmatically.

Also I think that horizontal `LinearLayout` would be better/more common way to implement this.

Reply:
I was thinking of doing this programmatically. But I suspected it will not work properly. Also preview will not work.

At least with the old large widget the layout did not work correctly when using the programmatical approach.

LinearLayout did not work properly with the background bars here. Therefore I did RelativeLayout here.

I would like to postpone this issue. This will take too much time now.
It took a lot of time already to get this somewhat looking OK.

You can create a ticket if you want.

Reply:
I think that it was working properly in old widget. There was just issue that amount of timesteps was not constant and it didn't work perfectly with all screen resolutions. Now we have constant amount of timesteps (5) so it is easier to implement.

But we can postpone this and I can fix this when iOS widgets are ready.